src/aot/aot.cpp

```diff
@@ -253,16 +256,62 @@ int load(BPFtrace &bpftrace, const std::string &in)
     return 1;
   }
 
-  uint8_t *ptr = static_cast<uint8_t *>(
-      ::mmap(NULL, in_file_size, PROT_READ, MAP_PRIVATE, infd, 0));
-  if (ptr == MAP_FAILED) {
-    auto saved_err = errno;
-    LOG(ERROR) << "Failed to mmap: " << in << ": " << std::strerror(saved_err);
-    return 1;
+  // Find .btaot section
+  Elf *elf = NULL;
+  Elf_Scn *scn = NULL;
+  GElf_Shdr shdr;
+  char *secname = NULL;
+  Elf_Data *data = NULL;
+  uint8_t *btaot_section = NULL;
+  const Header *hdr;
+  int i = 0;
```

Comment:
Yeah, I put it here (along with the others) because of the goto statements bypassing initialization errors. But I can split the initialization onto a second line to make it work.